fix(lib/solvers/glpk): parse file following GLPK LP/MIP format

kyrantonia opened the pull request:
## Contexto
Rulp es una librería que permite obtener soluciones a problemas de programación lineal (LP) y programación entera mixta (MIP) facilitando la generación del archivo con el problema y la lectura del archivo con la solución.
Para utilizarla se debe instalar un solver, para el caso de Platanus se utilizar GLPK en la versión 4.64.
## Problema
Se encontró un bug utilizando la librería que generaba que el archivo de respuesta no se leyera correctamente.
## Solución
Se cambiará la manera en que se lee la respuesta utilizando el formato estándarizado llamado GLPK LP/MIP. La documentación aparece en el [Manual de referencia GNU Linear Programming Kit](https://hpc.nih.gov/apps/glpk/latest/glpk.pdf) en el capitulo _Utility API routines_ secciones Problem data reading/writing routines y Problem solution reading/writing routines.

## Qué se está haciendo
- Se agregan las flags --write y --wglp para obtener la solución y el problema respectivamente en este formato.
- Se lee el archivo del problema para obtener el id interno de las variables del problema original. Esto sigue el formato `n j COL NAME` (ver foto Problema nuevo formato).
- Se lee el archivo solución hasta llegar a la linea correspondiente a la solución la cual contiene el estado de ésta (SST o PST), el tipo de problema(bas, ipt o mip) y la respuesta para la función objetivo (OBJ). Se tienen los siguientes formatos dependiendo del tipo de problema:
`s bas ROWS COLS PST DST OBJ`
`s ipt ROWS COLS SST OBJ`
`s mip ROWS COLS SST OBJ`
- Se lee nuevamente el archivo solución pero esta vez completo para obtener las respuestas de las variables (PRIM o VAL). Se tienen los siguientes formatos dependiendo del tipo de problema:
`j COL ST PRIM DUAL`
` j COL PRIM DUAL`
` j COL VAL`
- Finalmente utilizando los ids internos, se le asigna a las variables entregadas en `store_results` sus respectivas soluciones.
## Otros
Solución formato antiguo
![image](https://user-images.githubusercontent.com/19381243/133497290-edcbb989-4910-4447-8e5b-63f0fd9e14e6.png)
Problema nuevo formato
![image](https://user-images.githubusercontent.com/19381243/133497583-f4deed04-cc50-4195-a87d-b4c0412251c4.png)
Solución nuevo formato
![image](https://user-images.githubusercontent.com/19381243/133497396-729f106b-ac36-4f47-8adf-64f83029979c.png)

kyrantonia committed:
fix(lib/solvers/glpk): parse file following GLPK LP/MIP format

rjherrera commented on lib/solvers/glpk.rb:
```diff
@@ -1,6 +1,8 @@
   def solve
+    glpk_filename = @outfile.chomp(".sol")
     command = "#{executable} --lp #{@filename} %s --cuts --output #{@outfile}"
+    command %= "--write #{glpk_filename}glpk.sol --wglp #{glpk_filename}glpk.prob"
```
esto no es opcional? como los de abajo que hace `options[:gap] ? "--...."`? y no cacho muy bien si tiene sentido o no, pero podría ir dentro de ese arreglo

kyrantonia replied:
ah no no es opcional, debería ponerla arriba :scream:

rjherrera commented on lib/solvers/glpk.rb:
```diff
@@ -12,22 +14,53 @@ def self.executable
+    column_solutions = get_column_solutions(sol_lines, solution_format)
     variables.each do |var|
-      var.value = vars_by_name[var.to_s].to_f
+      id = column_ids[var&.to_s]
```
cuando es nil la var qué pasa? no importa que var.value al final quede nil?

kyrantonia replied:
mm creo que var no puede ser nil, quizas en ese caso es mejor sacarle el & para que no quede raro :thinking:

rjherrera reviewed (commented):
ahí te puse unos comentarios! se ve bien pero en verdad mucho no entiendo jajaja

difernandez committed:
fix(gemspec): add missing require date

difernandez committed:
fix(gemfile): upgrade pry-byebug to fix error on bundle install

difernandez committed:
fix(): require Rulp:: prefix to define variables

Previously, a monkeypatch to Kernel#method_missing and Object#const_missing was used. The first one was causing significant app slowing in development, so this attempts to fix that

difernandez committed:
Merge pull request #3 from platanus/fix-kernel-method-missing-slowing-apps

Fix kernel method missing slowing apps
